### PR TITLE
Multiword response v2 integration test

### PR DESCRIPTION
**Pull request by spooktheducks**

### CODE REVIEW

File: core/services/pipeline/common.go [hunk 1]
```diff
@@ -174,12 +174,12 @@ func (js *JSONSerializable) UnmarshalJSON(bs []byte) error {
 }
 
 func (js JSONSerializable) MarshalJSON() ([]byte, error) {
-	switch x := js.Val.(type) {
-	case []byte:
-		return json.Marshal(string(x))
-	default:
-		return json.Marshal(js.Val)
-	}
+	// switch x := js.Val.(type) {
```
Comment (spooktheducks) [inline core/services/pipeline/common.go:177]: Does anyone remember why we forced `[]byte` to `string` here?

File: core/services/pipeline/common_eth.go [hunk 2]
```diff
@@ -90,12 +96,173 @@ func parseETHABIString(theABI []byte, isLog bool) (name string, args abi.Argumen
 		return "", nil, nil, errors.Errorf("bad ABI specification: %v", theABI)
 	}
 	name = string(bytes.TrimSpace(matches[0][1]))
-	if len(name) == 0 {
```
Comment (spooktheducks) [inline core/services/pipeline/common_eth.go:93]: This checks to make sure that the `abi` attribute contains the log name, but there was no functional value in doing so. Not married to removing it, though -- happy to add it back if people prefer

File: core/internal/cltest/cltest.go [hunk 3]
```diff
@@ -1397,6 +1399,13 @@ func AssertPipelineRunsStays(t testing.TB, pipelineSpecID int32, store *strpkg.S
 	return prs
 }
 
+func AssertPipelineTaskRunsSuccessful(t testing.TB, runs []pipeline.TaskRun) {
+	t.Helper()
+	for _, run := range runs {
+		require.True(t, run.Error.IsZero())
```
Comment (j16r) [inline core/internal/cltest/cltest.go:1405]: `require.NoError` would be better here, no?

Comment (j16r) [reply core/internal/cltest/cltest.go:1405]: I'd also suggest maybe adding a message to indicate which of the pipeline task runs failed, including the index in the array, the name, etc.

File: core/services/pipeline/common_eth.go [hunk 4]
```diff
@@ -90,12 +96,173 @@ func parseETHABIString(theABI []byte, isLog bool) (name string, args abi.Argumen
...
-		if isLog {
-			return "", nil, nil, errors.Errorf("bad ABI specification: no event name")
-		}
-		return "", nil, nil, errors.Errorf("bad ABI specification: no method name")
-	}
 	args, indexedArgs, err = parseETHABIArgsString(matches[0][2], isLog)
 	return name, args, indexedArgs, err
 }
+
+func convertToETHABIType(val interface{}, abiType abi.Type) (interface{}, error) {
```
Comment (j16r) [inline core/services/pipeline/common_eth.go:103]: Is this intended to mimic the old behavior?

File: core/services/pipeline/task.eth_abi_encode.go [hunk 5]
```diff
@@ -53,9 +57,10 @@ func (t *ETHABIEncodeTask) Run(_ context.Context, vars Vars, inputs []Result) (r
 		if !exists {
 			return Result{Error: errors.Wrapf(ErrBadInput, "ETHABIEncode: argument '%v' is missing", arg.Name)}
 		}
-		val, err = convertToABIType(val, arg.Type)
+		logger.Warnf("  - %v: (%T / %v) %v", arg.Name, val, arg.Type.GetType(), val)
```
Comment (j16r) [inline core/services/pipeline/task.eth_abi_encode.go:60]: This looks a little hard to grep for, is there a human component we could add?

**Review by j16r — APPROVED** [review]
Looks good to me, added a couple of nitpicks / questions.

File: core/services/pipeline/common_eth.go (hunk 4 above)
Comment (spooktheducks) [reply core/services/pipeline/common_eth.go:103]: It's intended to be a bit more expansive, particularly given that we're doing multiword responses now. It supports all sorts of exotic types, such as `[][69]bool` and `[420][]address` (those are Optimism chain IDs, don't get any ideas)

File: core/internal/cltest/cltest.go (hunk 3 above)
Comment (spooktheducks) [reply core/internal/cltest/cltest.go:1405]: Normally yeah, but this is not actually a Go `error` type -- it's a `null.String`

File: core/services/pipeline/task.eth_abi_encode.go (hunk 5 above)
Comment (spooktheducks) [reply core/services/pipeline/task.eth_abi_encode.go:60]: Whoops, didn't mean for this to be included, will remove

File: core/services/pipeline/common_eth.go (hunk 4 above)
Comment (j16r) [reply core/services/pipeline/common_eth.go:103]: Is it backwards compatible, do we need the script to take care of this if not?

**Review by j16r — DISMISSED** [review]

**Review by archseer — DISMISSED** [review]
Looked at it earlier and it looked fine. Didn't approve since the tests failed but it looks like it's just a flake.

**Review by archseer — APPROVED** [review]

**Pull request merged**